### PR TITLE
feat(unlock-app) - Settings transfer fees

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Screenshot 2022-11-24 at 15 19 19

Screenshot 2022-11-24 at 15 08 34

Screenshot 2022-11-24 at 15 23 04

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/UpdateTransferFee.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const updateTransferFeeMutation = useMutation(updateTransferFee)
+
+  const onSubmit = async (fields: FormProps) => {
+    if (isValid) {
+      const updateTransferFeePromise =
+        updateTransferFeeMutation.mutateAsync(fields)
+
+      await ToastHelper.promise(updateTransferFeePromise, {
+        loading: 'Updating transfer fee',
+        error: 'Impossible to update transfer fee.',
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/forms/UpdateTransferFee.tsx]: added a more generic message

[the remark this answers is not in the document]
Comment (searchableguy) [reply unlock-app/src/components/interface/locks/Settings/forms/UpdateTransferFee.tsx]: I would remove impossible. Failed to update seems more appropriate

**Review by searchableguy — APPROVED** [review]

File: packages/unlock-js/src/walletService.ts [hunk 2]
```diff
@@ -868,7 +868,31 @@ export default class WalletService extends UnlockService {
...
   /**
-   * Upgrade a lock to a specific version
+   * Allow a Lock manager to change the transfer fee.
+   * @param {*} params
+   * @param {*} callback
+   */
+  async updateTransferFee(
+    params: {
+      lockAddress: string
+      transferFeeBasisPoints: number
```
Comment (searchableguy) [inline packages/unlock-js/src/walletService.ts:878]: should we allow values in percentage here? Seems more friendly than using basis points.

Comment (kalidiagne) [reply packages/unlock-js/src/walletService.ts:878]: but is stores the basis point not the percentage value, can cause confusion

**Pull request merged**